invenio_app/config.py

```diff
@@ -18,22 +18,89 @@
 websites.
 """
 
-RATELIMIT_DEFAULT = '5000/hour'
-"""Default rate limit.
+from __future__ import absolute_import, print_function, unicode_literals
 
-.. note:: Overwrite
-   Flask-Limiter <https://flask-limiter.readthedocs.io/en/stable/>`_
-   configuration.
+from invenio_app.limiter import set_rate_limit
+
+RATELIMIT_APPLICATION = set_rate_limit
+"""Global rate limit."""
+
+RATELIMIT_STRATEGY = 'moving-window'
+"""The rate limiting strategy to use.
+
+The strategy used here is the most consistant but also expensive one.
+If you are experiencing performance issues due to the increased Redis
+traffic, you can replace it with another one from the following
+`Flask-Limiter strategies
+<https://flask-limiter.readthedocs.io/en/stable/#ratelimit-strategy>`_.
 """
 
 RATELIMIT_HEADERS_ENABLED = True
-"""Enable rate limit headers. (Default: ``True``)
+"""Enable rate limit headers. (Default: ``True``)"""
 
-.. note:: Overwrite
-   Flask-Limiter <https://flask-limiter.readthedocs.io/en/stable/>`_
-   configuration.
+RATELIMIT_STORAGE_URL = 'memory://'
+"""Provide your Redis URL if you are rate limiting a multithreaded application.
+
+.. note::
+
+    Memory is used by default if no value is provided.
+    For more information regarding the mentioned above configuration values and
+    their available options you can see the `Flask-Limiter configuration
+    <https://flask-limiter.readthedocs.io/en/stable/#configuration>`_.
+"""
+
+RATELIMIT_KEY_FUNC = None
+"""Define custom key function.
+
+This config is not part of Flask-Limiter.
+If defined it will override the predefined ``key_func``,which is creating the
+key by attaching the user agent to the IP address of the user.
+This function is used to generate a unique key for each visitor to track
+the number of their requests.
+For more information you can also see `here
+<https://flask-limiter.readthedocs.io/en/stable/#rate-limit-domain>`_
 """
 
+RATELIMIT_PRIVILEGED_USERS = {}
+"""Specifically defined Flask rate limits per user.
+
+This config is not part of Flask-Limiter.
+It is used to apply custom rate limits to the included users, by providing the
+user's email accompanied by a `rate limit value
+<https://flask-limiter.readthedocs.io/en/stable/#rate-limit-string-notation>`_.
+
+.. code-block:: python
+
+    RATELIMIT_PRIVILEGED_USERS = \
+    {
+        'privileged.user@cern.ch': '6000 per hour;200 per minute',
+        'admin@cern.ch': '1000 per minute'
+    }
+"""
+
+RATELIMIT_SPECIFIC_ENDPOINTS = {}
```

Comment:
If you would go for something like `RATELIMIT_PER_USER_LIMIT ` for the previous variable, something like `RATELIMIT_PER_ENDPOINT_LIMIT ` would make sense since in an abstract way "both do the same" and have the same data structure, so it would be more harmonised.